cypress-io/cypress
# chore: add better-sqlite3 dependency

### Additional details


This PR coincides with the services PR here: https://github.com/cypress-io/cypress-services/pull/5467

This PR adds [`better-sqlite3`](https://github.com/WiseLibs/better-sqlite3) as a dependency.

The dependency needs to be built for the proper version of electron that Cypress uses, so we set up the rebuild process on post install and binary build.

### Steps to test


- Clone the `cypress-services` repo and checkout `ryanm/chore/better-sqlite3`
  - Run `yarn`
  - Run `yarn watch` in `packages/app-capture-protocol`
- Set `CYPRESS_LOCAL_PROTOCOL_PATH` to the path to `cypress-services/packages/app-capture-protocol/dist/index.js`
- Clone the `cypress` repo  and checkout `ryanm/chore/better-sqlite3`
  - Run `yarn`
  - Execute `cypress run` on a project in record mode with `DEBUG` set to `cypress:*protocol*` and ensure you can see the lifecycle events

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
